ui/components/app/modals/account-details-modal/account-details-modal.js

```diff
@@ -0,0 +1,146 @@
+import React, { useContext, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { getAccountLink } from '@metamask/etherscan-link';
+import { useHistory } from 'react-router-dom';
+
+import AccountModalContainer from '../account-modal-container';
+import QrView from '../../../ui/qr-code';
+import EditableLabel from '../../../ui/editable-label';
+import Button from '../../../ui/button';
+import { getURLHostName } from '../../../../helpers/utils/util';
+import { isHardwareKeyring } from '../../../../helpers/utils/hardware';
+import { EVENT } from '../../../../../shared/constants/metametrics';
+import { NETWORKS_ROUTE } from '../../../../helpers/constants/routes';
+import {
+  curriedGetEnsResolutionByAddress,
+  getBlockExplorerLinkTextForAccountDetailsModal,
+  getCurrentChainId,
+  getMetaMaskAccountsOrdered,
+  getMetaMaskKeyrings,
+  getRpcPrefsForCurrentProvider,
+  getSelectedIdentity,
+} from '../../../../selectors';
+import {
+  hideModal,
+  setAccountLabel,
+  showModal,
+  tryReverseResolveAddress,
+} from '../../../../store/actions';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { MetaMetricsContext } from '../../../../contexts/metametrics';
+
+export default function AccountDetailsModal() {
+  const { name, address } = useSelector(getSelectedIdentity);
+  const ensName = useSelector(curriedGetEnsResolutionByAddress(address));
+  const keyrings = useSelector(getMetaMaskKeyrings);
+  const chainId = useSelector(getCurrentChainId);
+  const accounts = useSelector(getMetaMaskAccountsOrdered);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const blockExplorerLinkText = useSelector(
+    getBlockExplorerLinkTextForAccountDetailsModal,
+  );
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  useEffect(() => {
+    if (address) {
+      dispatch(tryReverseResolveAddress(address));
+    }
+  }, [address, dispatch]);
+
+  const keyring = keyrings.find((kr) => {
+    return kr.accounts.includes(address);
+  });
+
+  const getAccountsNames = (allAccounts, currentName) => {
+    return Object.values(allAccounts)
+      .map((item) => item.name)
```

Comment:
I don't think we need to do two loops just to get the `name` --- that's a perf hit that's unnecessary.

Reply:
Good point.